sdk/clientcore/core/src/main/java/io/clientcore/core/utils/CodegenUtil.java

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package io.clientcore.core.utils;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+
+/**
+ * Utility class for code generation.
+ */
+public final class CodegenUtil {
+
+    /**
+     * Infer the type name from the return type string.
+     *
+     * @param returnTypeString The return type string.
+     * @return The inferred parameterized type.
+     * @throws RuntimeException If the class cannot be found.
+     */
+    public static ParameterizedType inferTypeNameFromReturnType(String returnTypeString) {
```

Comment:
This is a bit confusing to me on why this uses `String` for the return type. Shouldn't we know the full `Type` when generating from annotations?